hkmc2/shared/src/main/scala/hkmc2/codegen/HandlerLowering.scala

```diff
@@ -486,7 +489,7 @@ class HandlerLowering(paths: HandlerPaths)(using TL, Raise, Elaborator.State, El
     
     val resumedVal = VarSymbol(Tree.Ident("value$"))
 
-    def createAssignment(sym: Local) = Assign(sym, resumedVal.asPath, End())
+    def createAssignment(sym: Local) = Reassign(sym, resumedVal.asPath, End()) // TODO: or is this `Assign`?
```

Comment:
Cc @CAG2Mark @AnsonYeung

Reply:
This is always a reassignment, as the old symbol would hold the previous EffectSig